src/entities/image-proxy/image-proxy.ts

```diff
@@ -0,0 +1,203 @@
+
+    this.id = rawPayload.id
+    this.createdAt = rawPayload.created_at ? new Date(rawPayload.created_at) : undefined
+    this.updatedAt = rawPayload.updated_at ? new Date(rawPayload.updated_at) : undefined
+    this.deleted = rawPayload.deleted ?? false
+    this.active = rawPayload.active ?? true
+    this.name = rawPayload.name
+    this.isDefault = rawPayload.is_default
+    this.proxyVendor = rawPayload.proxy_vendor
+    this.configuration = rawPayload.configuration
+    this.labels = rawPayload.labels
+
+    return this
+  }
+
+  public static create (payload: ImageProxyRawPayload, universe: Universe, http: Universe['http']): ImageProxy {
+    return new ImageProxy({ rawPayload: payload, universe, http, initialized: true })
+  }
+
+  public serialize (): ImageProxyRawPayload {
+    return {
+      id: this.id,
+      created_at: this.createdAt ? this.createdAt.toISOString() : undefined,
+      updated_at: this.updatedAt ? this.updatedAt.toISOString() : undefined,
+      deleted: this.deleted ?? false,
+      active: this.active ?? true,
+      name: this.name,
+      is_default: this.isDefault,
+      proxy_vendor: this.proxyVendor,
+      configuration: this.configuration,
+      labels: this.labels
+    }
+  }
+
+  public async init (): Promise<ImageProxy | undefined> {
+    try {
+      await this.fetch()
+
+      return this
+    } catch (err) {
+      throw this.handleError(new ImageProxyErrorInitializationError(undefined, { error: err }))
+    }
+  }
+
+  public async auth (): Promise<boolean> {
+    if (!this.configuration) return false
+    try {
+      const opts = {
+        method: 'PUT',
+        url: `${this.apiCarrier?.injectables?.base}/api/v0/image_proxy/auth`,
+        headers: {
+          'Content-Type': 'application/json; charset=utf-8'
+        },
+        data: { },
+        responseType: 'json'
+      }
+
+      const response = await this.http?.getClient()(opts)
+
+      return response.headers.code === 200
+    } catch (error) {
+      throw new UrlShortenerImgProxyError({ error })
+    }
+  }
+
+  public getResizedImgUrl (request: ImageProxyResizeRequest): string {
+    if (!this.configuration) throw new TypeError('img proxy requires configuration to be set.')
+
+    return `${this.configuration.domain}/proxy/images?${queryString.stringify(this.getUrlParams(request))}`
+  }
+
+  private getUrlParams (request: ImageProxyResizeRequest): { [key: string]: string | number } {
+    const params = Object.assign({ uri: request.url },
+      request.dpr === null ? null : { dpr: request.dpr },
+      request.format === null ? null : { format: request.format },
+      request.background === null ? null : { background: request.background },
+      request.gravity === null ? null : { gravity: request.gravity }
+    )
+
+    if (request.resize) {
+      params['resize[type]'] = request.resize.type
+      params['resize[height]'] = request.resize.height
+      params['resize[width]'] = request.resize.width
+    }
```

Comment:
can't you not just use normal object notation and use `qs` for the stringified version?